core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -447,10 +449,28 @@ private[spark] class DAGScheduler(
       stageResourceProfiles: HashSet[ResourceProfile]): ResourceProfile = {
     logDebug(s"Merging stage rdd profiles: $stageResourceProfiles")
     val resourceProfile = if (stageResourceProfiles.size > 1) {
-      // add option later to actually merge profiles - SPARK-29153
-      throw new IllegalArgumentException("Multiple ResourceProfile's specified in the RDDs for " +
-        "this stage, please resolve the conflicting ResourceProfile's as Spark doesn't" +
-        "currently support merging them.")
+      if (shouldMergeResourceProfiles) {
+        var mergedProfile: ResourceProfile = stageResourceProfiles.head
+        for (profile <- stageResourceProfiles.drop(1)) {
+          mergedProfile = mergeResourceProfiles(mergedProfile, profile)
+        }
+        // compared merged profile with existing ones so we we don't add it over and over again
+        // if the user runs the same operation multiple times
+        val resProfile = sc.resourceProfileManager.getEquivalentProfile(mergedProfile)
+        resProfile match {
+          case Some(existingRp) => existingRp
```

Comment:
If user configured same resources for 2 separate stages, will these 2 stage using the same resource profile?
If so, will these 2 stages race for the resources?

Reply:
if a user configures same resource for 2 separate stages then yes it could use the same profile. This is no different than currently so I don't understand your question about race for resources.  You can run 2 stages now and they will be both require resources to run. The dynamic allocation manager handles this and asks for the appropriate number of executors. If you don't have enough to run all in parallel, the scheduler simply schedules as available.

Reply:
Yeah, I just want to make sure that my understanding is correct here by asking the question. Thanks for your explanation.